### PR TITLE
Rf08 tc001 tc ivan misericordia 07

### PR DESCRIPTION
**Pull request by ivanMise**
RF08-TC-001 UI >Reporte de Inventario > Verificar Cabecera de reporte de Inventario (Contenido e imagen)

image

image

### CODE REVIEW

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.umssdiplo.automationv01.core.managepage.BasePage;
+import org.umssdiplo.automationv01.core.utils.CommonEvents;
+
+public class ListInventory extends BasePage{
+
+    @FindBy(className = "mat-button mat-primary")
+    private WebElement aceptarBtn;
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java:11]: change to buttonAccept

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    WebElement headerReportImage;
+
+    @FindBy(xpath = "/html/body/ssi-root/ssi-inventory/div/div[2]/div/mat-card/mat-card-content/h1")
+    WebElement headerReportTitle;
+
+    public ListInventory() {
+        CommonEvents.isVisible(aceptarBtn);
+    }
+
+    public boolean isInventoryListVisible() {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java:26]: isVisibleInventoryList

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        return CommonEvents.isVisible(tableInventory);
+    }
+
+    public boolean checkHeaderReportInventory() {
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java:30]: isVisibleHeaderReportInventory

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuInventory.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.umssdiplo.automationv01.core.managepage.BasePage;
+import org.umssdiplo.automationv01.core.managepage.Equipment.ListInventory;
+import org.umssdiplo.automationv01.core.utils.CommonEvents;
+
+public class SubMenuInventory extends BasePage {
+    @FindBy(id = "smInventory")
+    private WebElement inventorySubMenu;
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuInventory.java:11]: subMenuInventory

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 5]
```diff
@@ -5,19 +5,24 @@
...
 import org.umssdiplo.automationv01.core.managepage.Login.Login;
 import org.umssdiplo.automationv01.core.managepage.Menu.Menu;
 import org.umssdiplo.automationv01.core.managepage.Menu.SubMenuEquipment;
 import org.umssdiplo.automationv01.core.managepage.Menu.SubMenuOrganizationalStructure;
 import org.umssdiplo.automationv01.core.managepage.Menu.SubMenuPersonal;
+import org.umssdiplo.automationv01.core.managepage.Menu.SubMenuInventory;
 import org.umssdiplo.automationv01.core.managepage.Position.Position;
 import org.umssdiplo.automationv01.core.managepage.Usuario.ListUser;
 import org.umssdiplo.automationv01.core.utils.ErrorMessage;
 import org.umssdiplo.automationv01.core.utils.LoadPage;
```
Comment (marbusjim) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java:23]: remove this line

File: src/test/resources/features/equipments/checkHeaderReportInventory.feature [hunk 6]
```diff
@@ -0,0 +1,12 @@
+Feature: Check Inventory Report Header
```
Comment (marbusjim) [inline src/test/resources/features/equipments/checkHeaderReportInventory.feature:1]: use spanish

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java (hunk 1 above)
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java:11]: change name webElement Ej: buttonAceptar

**Review by marbusjim — COMMENTED** [review]
with comments

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+public class ListInventory extends BasePage{
+
+    @FindBy(className = "mat-button mat-primary")
+    private WebElement aceptarBtn;
+
+    @FindBy(className = "mat-table")
+    private WebElement tableInventory;
+
+    @FindBy(xpath = "/html/body/ssi-root/ssi-inventory/div/div[2]/div/mat-card/mat-card-header")
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java:16]: change this xpath, more short and specific, Ej.: //mat-header-cell[contains(text(),'Cod. Personal')]

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    WebElement headerReportImage;
+
+    @FindBy(xpath = "/html/body/ssi-root/ssi-inventory/div/div[2]/div/mat-card/mat-card-content/h1")
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Equipment/ListInventory.java:19]: change this xpath, more short and specific

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/Menu.java [hunk 9]
```diff
@@ -39,8 +42,11 @@ public IncidentPage clickMenuIncident() {
         return new IncidentPage();
     }
 
-    public SubMenuEquipment selectEquipmentMenu() {
+    public List<BasePage> selectEquipmentMenu() {
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/Menu.java:45]: what is the purpose to obtain many PageObjects?

Comment (jdpapiero) [reply src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/Menu.java:45]: is better maintain the most atomic posible

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuInventory.java (hunk 4 above)
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuInventory.java:11]: remane webElement Ej.: subMenuInventory

File: src/test/resources/features/equipments/checkHeaderReportInventory.feature [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Verificar Cabecera de reporte de Inventario: Contenido e imagen
+    Given 'Menu principal' estee cargado
+    And seleccionar menu 'Equipamiento' en la pagina 'Menu Principal'
+    And Seleccionar submenu 'Inventory' en menu 'Equipamiento'
+    When 'Reporte de Inventarios' esta correctamente cargado
+    Then Verificar Cabecera de reporte de Inventario contenido-imagen
```
Comment (jdpapiero) [inline src/test/resources/features/equipments/checkHeaderReportInventory.feature:12]: first word of step in lowercase

File: src/test/resources/features/equipments/checkHeaderReportInventory.feature [hunk 11]
```diff
@@ -0,0 +1,12 @@
+Feature: Check Inventory Report Header
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Verificar Cabecera de reporte de Inventario: Contenido e imagen
+    Given 'Menu principal' estee cargado
+    And seleccionar menu 'Equipamiento' en la pagina 'Menu Principal'
+    And Seleccionar submenu 'Inventory' en menu 'Equipamiento'
```
Comment (jdpapiero) [inline src/test/resources/features/equipments/checkHeaderReportInventory.feature:10]: first word of step in lowercase

File: src/test/resources/features/equipments/checkHeaderReportInventory.feature (hunk 10 above)
Comment (jdpapiero) [reply src/test/resources/features/equipments/checkHeaderReportInventory.feature:12]: review Test Case writed for elements to validate

**Review by jdpapiero — CHANGES_REQUESTED** [review]
Review Observations

**Pull request closed without merging**